runtime/Cargo.toml

```
@@ -9,29 +9,55 @@ repository.workspace = true
 version.workspace = true
 
 [dependencies]
-wasmtime = {version = ">= 27.0.0, < 31.0.0"}
-wasi-common = {version = ">= 27.0.0, < 31.0.0"}
-wiggle = {version = ">= 27.0.0, < 31.0.0"}
+wasmtime = { version = ">= 27.0.0, < 31.0.0", default-features = false, features = [
+  'cache',
+  'gc',
+  'gc-drc',
+  'cranelift',
+  'coredump',
+  'wat',
+] }
+wasi-common = { version = ">= 27.0.0, < 31.0.0" }
+wiggle = { version = ">= 27.0.0, < 31.0.0" }
 anyhow = "1"
-serde = {version = "1", features = ["derive"]}
+serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 toml = "0.8"
 sha2 = "0.10"
 tracing = "0.1"
-tracing-subscriber = {version = "0.3.18", features = ["std", "env-filter", "fmt"]}
+tracing-subscriber = { version = "0.3.18", features = [
+  "std",
+  "env-filter",
+  "fmt",
+] }
 url = "2"
 glob = "0.3"
-ureq = {version = "3.0", optional=true}
+ureq = { version = "3.0", optional = true }
 extism-manifest = { workspace = true }
 extism-convert = { workspace = true, features = ["extism-path"] }
 uuid = { version = "1", features = ["v4"] }
 libc = "0.2"
 
 [features]
-default = ["http", "register-http", "register-filesystem"]
-register-http = ["ureq"] # enables wasm to be downloaded using http
-register-filesystem = [] # enables wasm to be loaded from disk
-http = ["ureq"]          # enables extism_http_request
+default = ["http", "register-http", "register-filesystem", "wasmtime-default-features"]
+register-http = ["ureq"]                                   # enables wasm to be downloaded using http
+register-filesystem = []                                   # enables wasm to be loaded from disk
+http = ["ureq"]                                            # enables extism_http_request
+wasmtime-default-features = [
+  "wasmtime/async",
+  "wasmtime/pooling-allocator",
+  'wasmtime/gc-null',
+  'wasmtime/profiling',
+  'wasmtime/parallel-compilation',
+  'wasmtime/pooling-allocator',
+  'wasmtime/demangle',
+  'wasmtime/addr2line',
+  'wasmtime/debug-builtins',
+  'wasmtime/runtime',
+  'wasmtime/component-model',
+  'wasmtime/threads',
+  'wasmtime/std',
+]
```

This approach is nice! I think we can do this instead:

```suggestion
wasmtime-default-features = [
  "wasmtime/default"
]
```

I will push up a commit with some small tweaks then will merge this!